tests/suites/test_suite_pkwrite.function

```diff
@@ -21,17 +22,30 @@ static void pk_write_check_common(char *key_file, int is_public_key)
     mbedtls_pk_init(&key);
     if (is_public_key) {
         TEST_ASSERT(mbedtls_pk_parse_public_keyfile(&key, key_file) == 0);
-        ret = mbedtls_pk_write_pubkey_pem(&key, buf, check_buf_len);
+        if (is_der) {
+            ret = mbedtls_pk_write_pubkey_der(&key, buf, check_buf_len);
+        } else {
+            ret = mbedtls_pk_write_pubkey_pem(&key, buf, check_buf_len);
+        }
     } else {
         TEST_ASSERT(mbedtls_pk_parse_keyfile(&key, key_file, NULL,
                                              mbedtls_test_rnd_std_rand, NULL) == 0);
-        ret = mbedtls_pk_write_key_pem(&key, buf, check_buf_len);
+        if (is_der) {
+            ret = mbedtls_pk_write_key_der(&key, buf, check_buf_len);
+        } else {
+            ret = mbedtls_pk_write_key_pem(&key, buf, check_buf_len);
+        }
     }
-    TEST_ASSERT(ret == 0);
 
-    /* check_buf_len also includes the NULL termination char */
-    TEST_EQUAL(check_buf_len - 1, strlen((char *) buf));
-    TEST_ASSERT(memcmp((char *) buf, (char *) check_buf, check_buf_len) == 0);
+    if (is_der) {
+        TEST_ASSERT(ret > 0);
+        start_buf = buf + check_buf_len - ret;
+    } else {
+        TEST_ASSERT(ret == 0);
+        start_buf = buf;
+    }
+
+    TEST_ASSERT(memcmp((char *) start_buf, (char *) check_buf, check_buf_len) == 0);
```

Comment:
Minor: we should be using `ASSERT_COMPARE()` in new code. This would ensure we're properly comparing the lengths too.